rerun-failed-run/rerun-failed-run.yml

```diff
@@ -0,0 +1,73 @@
+name: Retrigger failed run action
+
+on:
+  workflow_dispatch:
+    inputs:
+      run_id:
+        description: The id of the run which we want to retrigger
+        required: true
+      error_message:
+        description: Custom error message to look for in the logs.
+        required: true
+      repository:
+        description: Github repository
+        required: true
+jobs:
+  rerun:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Import Secrets
+        id: secrets
+        uses: hashicorp/vault-action@v2.7.4
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          secrets: |
+            secret/data/products/infra/ci/retrigger-gha-workflow RETRIGGER_APP_KEY;
+            secret/data/products/infra/ci/retrigger-gha-workflow RETRIGGER_APP_ID;
+
+      - name: Generate a GitHub token
+        id: github-token
+        uses: tibdex/github-app-token@v2
+        with:
+          app_id: ${{ steps.secrets.outputs.RETRIGGER_APP_ID }}
+          private_key: ${{ steps.secrets.outputs.RETRIGGER_APP_KEY }}
+
+      - name: Check previous run logs
+        id: check_logs
+        env:
+          GH_DEBUG: api
+        run: |
+          echo ${{ steps.github-token.outputs.token }} | gh auth login --with-token
+          echo "Fetching logs for run with id: ${{ inputs.run_id }} from ${{ inputs.repository }} repository."
+          # Fetch the logs from the previous run
+          logs=$(gh run view ${{ inputs.run_id }} -R ${{ inputs.repository }} --log)
+          echo "Logs coming from run with id: ${{ inputs.run_id }}"
+          echo "$logs"
+          echo "end logs"
+          # Check if the desired string is present in the logs
+          if echo "$logs" | grep -F "${{ inputs.error_message }}" >/dev/null; then
+            count=$(echo "$logs" | grep -F "${{ inputs.error_message }}" | wc -l)
+            echo "count: $count"
+            if [ "$count" -gt 2 ]; then
+                echo "::set-output name=rerun::true"
+                echo "Error message ${{ inputs.error_message }} found more than two times in logs so the previous run has finished with that error."
+            else
+                echo "::set-output name=rerun::false"
+                echo "Error message ${{ inputs.error_message }} found in logs, but not more than one time"
+            fi
+          else
+              echo "::set-output name=rerun::false"
+              echo "Error message $error_message not found in logs"
+          fi
+
+      - name: Rerun run with id ${{ inputs.run_id }} for ${{ inputs.repository }} repository
+        if: steps.check_logs.outputs.rerun == 'true'
+        env:
+          GH_DEBUG: api
+        run: |
+          echo ${{ steps.github-token.outputs.token }} | gh auth login --with-token
+          gh run watch ${{ inputs.run_id }} -R ${{ inputs.repository }} > /dev/null 2>&1
```

Comment:
shall this workflow be in generaly having a timeout safety net that won't let it run forever? the risk I see now if that `gh run watch` can be stuck if the original workflow is taking too long to finish.

Reply:
yeah I agree, imo it is maybe actually not necessary to watch the run now that I think about it again - maybe it is enough to only retrigger it